### ISSUE TITLE
node-param-description-miscased-json: false positive on n8n $json expression variable

### ISSUE THREAD

**Issue by SoapyRED** [ivov/eslint-plugin-n8n-nodes-base#209]
## Reproduction

Minimal failing case in `nodes/MyNode/MyNode.node.ts`:

```ts
{
  displayName: 'Items Source',
  name: 'itemsSource',
  type: 'options',
  default: 'list',
  description: 'Pass an array as a JSON expression (e.g. ={{ $json.items }})',
}
```

The string `={{ $json.items }}` is canonical n8n syntax for referencing the JSON output of the previous node — `$json` is a [runtime expression variable](https://docs.n8n.io/code/builtin/current-node-input/) and is required to be lowercase.

## Expected behaviour

Rule does not flag `$json` because it is an n8n expression-variable token, not a reference to the JSON format.

## Actual behaviour

`node-param-description-miscased-json` fires:

```
error  Use 'JSON' [autofixable]  n8n-nodes-base/node-param-description-miscased-json
```

`--fix` rewrites `$json.items` to `$JSON.items`, which is **broken n8n expression syntax** — n8n's expression engine looks up the lowercase `$json`. So the autofix actively breaks any description that documents the canonical `={{ $json.items }}` example for users.

## Cause

In `dist/lib/constants.js`:

```js
const MISCASED_JSON_REGEX = /\b(j|J)son\b/;
```

In `dist/lib/rules/node-param-description-miscased-json.js`:

```js
const correctlyCased = description.value.replace(/\bjson\b/gi, 'JSON');
```

Both rely on `\b` word boundaries. `$` is a non-word character, so `\b` matches the boundary between `$` and `json`, and the regex captures the `json` inside `$json`.

## Suggested fix

Add a negative lookbehind for `$` to both regexes so they skip n8n expression tokens:

```js
// constants.js
const MISCASED_JSON_REGEX = /(?<!\$)\b(j|J)son\b/;

// node-param-description-miscased-json.js
const correctlyCased = description.value.replace(/(?<!\$)\bjson\b/gi, 'JSON');
```

This still flags genuine references to the JSON format ("a json string", "JSON expression", etc.) but skips `$json`, `$JSON` (and would skip any future n8n `$<lowercase>` expression tokens — `$node`, `$workflow`, `$execution`, etc., none of which collide with this rule today but the same defensive pattern applies).

## Workaround for downstream consumers

File-scope rule disable:

```js
// eslint.config.mjs
export default [{
  files: ['nodes/**/*.node.ts'],
  rules: {
    'n8n-nodes-base/node-param-description-miscased-json': 'off',
  },
}];
```

This removes useful linting on actual JSON-format references in the same file. Suboptimal but unblocks CI.

## Environment

- `eslint-plugin-n8n-nodes-base`: 1.16.5
- `eslint`: 9.39.4
- `@n8n/node-cli`: latest (transitively pulls the plugin)
- Node: 18+ / 20+
- OS: Windows 11 (also reproduced on Ubuntu 24.04 in CI)

Hit on `n8n-nodes-freightutils` v0.3.0 release. Reproduction is downstream-agnostic — any community node whose description documents `={{ $json.* }}` triggers it.
